src/pages/ReimbursementAccount/ReimbursementAccountPage.js

```diff
@@ -214,129 +208,151 @@ class ReimbursementAccountPage extends React.Component {
+    );
+
+    const [shouldShowContinueSetupButton, setShouldShowContinueSetupButton] = useState(hasACHDataBeenLoaded ? getShouldShowContinueSetupButtonInitialValue() : false);
+
+    const currentStep = achData.currentStep || CONST.BANK_ACCOUNT.STEP.BANK_ACCOUNT;
+    const policyName = lodashGet(policy, 'name', '');
+    const policyID = lodashGet(route.params, 'policyID', '');
+    const {translate} = useLocalize();
+    const {isOffline} = useNetwork();
+    const requestorStepRef = useRef(null);
+    const prevReimbursementAccount = usePrevious(reimbursementAccount);
+    const prevIsOffline = usePrevious(isOffline);
 
     /**
      * Retrieve verified business bank account currently being set up.
      * @param {boolean} ignoreLocalCurrentStep Pass true if you want the last "updated" view (from db), not the last "viewed" view (from onyx).
      */
-    fetchData(ignoreLocalCurrentStep) {
+    function fetchData(ignoreLocalCurrentStep) {
         // Show loader right away, as optimisticData might be set only later in case multiple calls are in the queue
         BankAccounts.setReimbursementAccountLoading(true);
 
         // We can specify a step to navigate to by using route params when the component mounts.
         // We want to use the same stepToOpen variable when the network state changes because we can be redirected to a different step when the account refreshes.
-        const stepToOpen = this.getStepToOpenFromRouteParams();
-        const achData = lodashGet(this.props.reimbursementAccount, 'achData', {});
+        const stepToOpen = getStepToOpenFromRouteParams(route);
         const subStep = achData.subStep || '';
         const localCurrentStep = achData.currentStep || '';
         BankAccounts.openReimbursementAccountPage(stepToOpen, subStep, ignoreLocalCurrentStep ? '' : localCurrentStep);
     }
 
-    continue() {
-        this.setState({
-            shouldShowContinueSetupButton: false,
-        });
-        this.fetchData(true);
+    // Will run whenever reimbursementAccount prop changes, and update the hasACHDataBeenLoaded state if necessary
+    useEffect(() => {
+        setHasACHDataBeenLoaded(reimbursementAccount !== ReimbursementAccountProps.reimbursementAccountDefaultProps);
+    }, [reimbursementAccount]);
+
+    useEffect(
+        () => {
+            fetchData();
+            return () => {
+                BankAccounts.clearReimbursementAccount();
+            };
+        },
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+        [],
+    ); // The empty dependency array ensures this runs only once after the component mounts.
+
+    // Check for network change from offline to online
+    if (prevIsOffline && !isOffline && prevReimbursementAccount && prevReimbursementAccount.pendingAction !== CONST.RED_BRICK_ROAD_PENDING_ACTION.DELETE) {
+        fetchData();
```

Comment:
Is it a bit weird that this is happening outside of a `useEffect()`. Can we move it inside one?

Reply:
@marcaaron I was asked to move it outside the `useEffect`. Please decide together if you like to place this code inside the useEffect or outside as it is now.

Reply:
Sure no worries. I moved it back inside the `useEffect` and pushed the changes.

Reply:
Sure nom worries. I moved it back inside the `useEffect`.

Reply:
Mainly asked because I think it is more common to see the side effects like this inside the `useEffect()`. And making an API call in the render felt unusual to me.